dcpy/connectors/edm/publishing.py

```diff
@@ -283,11 +283,15 @@ def validate_or_patch_version(
 
     if version_already_published:
         if is_patch:
-            return versions.bump(
+            patched_version = versions.bump(
                 previous_version=version,
                 bump_type=versions.VersionSubType.patch,
                 bump_by=1,
             ).label
+            assert patched_version not in get_published_versions(
```

Comment:
nit: should we raise a ValueError here? (just to maintain consistency in what Exceptions this fn raises)